guardrails/validators.py

```diff
@@ -2558,7 +2589,15 @@ def get_modified_value(
     def validate(self, value: str, metadata: Dict[str, Any]) -> ValidationResult:
         # Check if value is a multiline string
         if "\n" not in value:
-            raise ValueError("You must provide a multiline code snippet.")
+            # Raise warning if value is not a multiline string
+            warnings.warn(
+                "You must provide a multiline code snippet in order to "
+                "use the DetectSecrets validator. "
```

Comment:
I would tweak this to something like `The DetectSecrets validator works best with multiline code snippets.`. Next line pointing to docs is good though.